frontend/plugins/resources/src/components/LibrariesPage/LibrariesPage.tsx

```diff
@@ -1,13 +1,108 @@
 import React from 'react';
-import { Content, Page } from '@backstage/core';
+import { Content, Page, Table, TableColumn, TableFilter} from '@backstage/core';
+import { getResources, DummyResponse, Resource, Edge } from 'zoo-api';
+// import {useQuery} from "urql";
+import {Chip, Grid} from "@material-ui/core";
+import Alert from '@material-ui/lab/Alert';
 
+import ResourceKind from "../TableComponents/ResourceKind";
+import ResourceTypeLabel from "../TableComponents/ResourceTypeLabel";
+import ResourceLink from "../TableComponents/ResourceLink";
 
-const LibrariesPage = () => (
-  <Page themeId="home">
-    <Content>
-      <div>Welcome to libraries page!</div>
-    </Content>
-  </Page>
-);
+
+const generateTableData: (resources: Array<Resource>) => Array<{}> = (resources) => {
+  const data: Array<{}> = [];
+  for(let i=0; i<resources.length; i++){
+    data.push({
+      usage: resources[i].usageCount,
+      name: <ResourceLink name={resources[i].name} id={resources[i].id} />,
+      version: <Chip label={resources[i].version} />,
+      type: <ResourceTypeLabel name={resources[i].type} />,
+      kind: <ResourceKind name={resources[i].name} />,
+    });
+  }
+
+  return data;
```

Comment:
Same comment regarding `return resources.map()` :arrow_up:

Reply:
Fixed, i have used map as you suggested. It is cleaner